Remove `page_title` attribute from frontmatter in resource documentation, update external link

SarahFrench opened the pull request:
<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->

This PR removes a line from the template used to generated markdown files documenting MMv1 resources. Currently the template adds a `page_title` attribute in the YAML frontmatter of each resource's markdown file, but this field is not required for resource documentation. [The documentation for provider documentation](https://developer.hashicorp.com/terraform/registry/providers/docs#supported-attributes) describes how this attribute is only needed for files in the `/guides` folder. Those files are all handwritten and not related to this template, so it's safe to remove this field from the template.

This PR also updates a link in the `Timeouts` section in the docs ([example here](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/app_engine_service_split_traffic#timeouts)) . The template currently includes a link to the old documentation that was hosted at terraform.io. Although the old link is redirected to the new developer.hashicorp.com website, the docs have changed and the link doesn't send users to a relevant page.


___
<!--
Replace each [ ] with [X] to check it. Switch to the preview view to make it easier to click on links.
These steps will speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.
-->
If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- ~~[x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).~~
- [x] ~~[Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).~~
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none
    
Unless you choose release-note:none, please add a release note.

See .ci/RELEASE_NOTES_GUIDE.md for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

SarahFrench commented on mmv1/templates/terraform/resource.html.markdown.erb:
```diff
@@ -173,7 +172,7 @@ In addition to the arguments listed above, the following computed attributes are
 
 This resource provides the following
-[Timeouts](/docs/configuration/resources.html#timeouts) configuration options:
+[Timeouts](https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/retries-and-customizable-timeouts) configuration options:
```
Can't use the relative link anymore as docs have moved away from the terraform.io domain

SarahFrench commented:
Realised there are a ton of other templates - marking as draft until I've found them all

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7151-old..auto-pr-7151) ( 363 files changed, 296 insertions(+), 659 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7151-old..auto-pr-7151) ( 363 files changed, 296 insertions(+), 659 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-7151-old..auto-pr-7151) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7151-old..auto-pr-7151) ( 632 files changed, 296 insertions(+), 928 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7151-old..auto-pr-7151) ( 632 files changed, 296 insertions(+), 928 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-7151-old..auto-pr-7151) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7151-old..auto-pr-7151) ( 632 files changed, 374 insertions(+), 1006 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7151-old..auto-pr-7151) ( 632 files changed, 374 insertions(+), 1006 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-7151-old..auto-pr-7151) ( 2 files changed, 3 insertions(+), 3 deletions(-))

SarahFrench commented:
This PR is similar to [this past PR](https://github.com/GoogleCloudPlatform/magic-modules/pull/6238) that also cleaned up usage of frontmatter in the provider documentation.

Sorry that there are a lot of file changes (here and in the downstream). If it helps with reviewing this PR, the changes to the handwritten files were achieved using a global search and replace, where I searched with this regex (`\npage_title: "Google: google_[a-z_]+"`) and replaced with nothing.

shuyama1 reviewed (approved):
LGTM! Thanks!
